stan/math/prim/functor/apply_scalar_unary.hpp

```diff
@@ -143,8 +141,8 @@ struct apply_scalar_unary<F, std::vector<T>> {
    * Return type, which is calculated recursively as a standard
    * vector of the return type of the contained type T.
    */
-  using return_t =
-      typename std::vector<typename apply_scalar_unary<F, T>::return_t>;
+  using return_t = typename std::vector<
+      plain_type_t<typename apply_scalar_unary<F, T>::return_t>>;
```

Comment:
Is it true we'd never want to return expressions here? It makes sense just clarifying.

Reply:
I can't think of an example where returning a std::vector of expressions would make sense. But maybe there is one and we will change that later.